Source/WebKit/UIProcess/WebPageProxy.messages.in

```diff
@@ -379,7 +379,7 @@ messages -> WebPageProxy {
 #endif
 #if PLATFORM(MAC)
     # Autocorrection messages
-    ShowCorrectionPanel(enum:uint8_t WebCore::AlternativeTextType panelType, WebCore::FloatRect boundingBoxOfReplacedString, String replacedString, String replacementString, Vector<String> alternativeReplacementStrings)
+    ShowCorrectionPanel(enum:uint8_t WebCore::AlternativeTextType panelType, WebCore::FloatRect boundingBoxOfReplacedString, String replacedString, String replacementString, Vector<String> alternativeReplacementStrings, std::optional<WebCore::FrameIdentifier> rootFrameID)
```

Comment:
I think this parameter shouldn't be optional.  If we don't have a root frame ID, we are disconnected from the frame tree and should early return.

Reply:
I designed it to use an optional based on `convertRectToMainFrameCoordinates`, which take an optional frame identifier.

```
void WebPageProxy::convertRectToMainFrameCoordinates(WebCore::FloatRect rect, std::optional<WebCore::FrameIdentifier> frameID, CompletionHandler<void(std::optional<WebCore::FloatRect>)>&& completionHandler)
```

Reply:
At this point, though, it shouldn't be optional.  Same with AlternativeTextClient.  If we don't have a root frame ID, we shouldn't be making the AlternativeTextClient call.